### ISSUE TITLE
Magit buffers not showing in helm mini list when Ctrl-x b

### ISSUE THREAD

**Issue by jeremymeng** [syl20bnr/spacemacs#12837]
#### Description :octocat:
magit buffers are not shown in results of `helm-buffers-list)` 

#### Reproduction guide :beetle:
- Start Emacs
- Start `magit` on some repo
- `Ctrl-x b`

*Observed behaviour:* :eyes: :broken_heart:
magit status buffer is not in the helm list
![image](https://user-images.githubusercontent.com/7583839/66854106-89759380-ef35-11e9-8983-bc38c1d8ecdd.png)

*Expected behaviour:* :heart: :smile:
magit status buffers should be in the helm list

#### System Info :computer:
- OS: windows-nt
- Emacs: 26.2
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 59c95e7da)
- Graphic display: t
- Distribution: spacemacs
- Editing style: emacs
- Completion: helm
- Layers:
```elisp
(auto-completion docker emacs-lisp ibuffer git github helm html javascript markdown restclient search-engine spell-checking twitter typescript yaml)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS MODULES THREADS LCMS2

#### Backtrace :paw_prints:
```
<<BACKTRACE IF RELEVANT>>
```

**Comment by jeremymeng** [comment]
some how my helm-projectile.el was corrupted

**Comment by duianto** [comment]
It's not on your end, it was changed in this commit:
completion: filter helm-mini with useful/useless buffers regexp https://github.com/syl20bnr/spacemacs/commit/22713541f8276023f3bb3f3f4cbd895f49de8945

A new key binding was added in the following commit:
completion: add command for unfiltered buffers list under SPC b U https://github.com/syl20bnr/spacemacs/commit/68579144416dc4c9f21afc53b08267d513646ce8

These issues seem to be discussing the reason for the change.
What's the deal with "useful" and "useless" buffers? #6683
Tweaks to useful/useless buffers behavior #11545

**Comment by jeremymeng** [comment]
All these issues mentioned above are 2018 or earlier while I only started seeing this recently. I have been using develop branch so I am not sure it's the same old issue or something new.

**Comment by duianto** [comment]
Your right the commit dates are from when the PR was created:
Filter useless buffers in Helm-mini, add command to unfiltered list buffers #11555

But they were merged to the Spacemacs develop branch on Sept 8, 2019:
This develop branch commits page shows when they were merged to the develop branch:
https://github.com/syl20bnr/spacemacs/commits/develop?after=320504dfc157f292cf54cf5c76f9965bf5010042+104

Scroll near the bottom or search for:
`Commits on Sep 8, 2019`

**Comment by jeremymeng** [comment]
Ah got it. Thanks a lot. Now I know what to customize.

**Comment by hansbogert** [comment]
> Ah got it. Thanks a lot. Now I know what to customize.

what should you change to get magit buffers in the normal buffer list again? I've changed the `helm-boring-[something]` variable, but it seems the magit layer just hard-sets it to be ignored on restart.

**Comment by voidfoo** [comment]
@hansbogert 
```elisp
(setq spacemacs-useless-buffers-regexp '("magit-.*: .*")
```
in my `dotspacemacs/user-config`

**Comment by hansbogert** [comment]
> @hansbogert
> 
> ```emacs-lisp
> (setq spacemacs-useless-buffers-regexp '("magit-.*: .*")
> ```
> 
> in my `dotspacemacs/user-config`

@voidfoo 
even then this happens:

![image](https://user-images.githubusercontent.com/2630900/108499791-25907700-72af-11eb-95f3-212cf1494d0f.png)

**Comment by jeremymeng** [comment]
You should not have `magit: .*` there if you want to see magit status buffer etc.

**Comment by hansbogert** [comment]
@jeremymeng  I know, but it keeps getting reset to the above at startup of spacemacs. Should've sent more context in the image, Customize also says it's changed outside Customize

![image](https://user-images.githubusercontent.com/2630900/108504195-cb46e480-72b5-11eb-87bf-0a67729425cc.png)

**Comment by duianto** [comment]
It doesn't work to set the helm boring buffer variable directly:
❌ `(setq helm-boring-buffer-regexp-list '("magit-.*: .*"))`

because the `spacemacs-useless-buffers-regexp` variables contents are appended to the helm boring variable when helm is loaded.
https://github.com/syl20bnr/spacemacs/blob/e4452f9a855d452aa3969ba8a5268f27ba936b06/layers/%2Bcompletion/helm/packages.el#L169-L171

`voidfoo`'s suggestion works by adding the following to the `dotspacemacs/user-config` section:
✔️ `(setq spacemacs-useless-buffers-regexp '("magit-.*: .*"))`

After restarting:
```
spacemacs-useless-buffers-regexp is a variable defined in ‘config.el’.
Its value is ("magit-.*: .*")
```

- Opening the HELM Projectile buffer: `SPC p p` 
- Selecting `~/.emacs.d`
- Pressing `C-z` shows the helm actions, which lists:
``Open project root in vc-dir or magit `M-g'``
- Pressing `C-z` again to returns to the HELM Projectile buffer
- Pressing: `M-g` opens the `magit: .emacs.d` buffer.
- Pressing: `SPC b b` lists the magit buffer in the HELM Mini buffer.
- Pressing: `$` (`magit-process-buffer`) opens the `magit-process: .emacs.d` buffer.
It doesn't appear in the `SPC b b` buffer (as expected), 
but it appears in the `SPC b U` (`helm-buffers-list-unfiltered`) HELM Buffers buffer. (also expected).

The helm boring buffer variable shows:
```
helm-boring-buffer-regexp-list is a variable defined in ‘helm-buffers.el’.
Its value is
("\\` " "\\`\\*helm" "\\`\\*Echo Area" "\\`\\*Minibuf" "magit-.*: .*")
Original value was 
("\\` " "\\`\\*helm" "\\`\\*Echo Area" "\\`\\*Minibuf")
```
And the `customize` buffer shows:
```
Hide Helm Boring Buffer Regexp List:
INS DEL Choice: Value Menu Regexp: \` 
INS DEL Choice: Value Menu Regexp: \`\*helm
INS DEL Choice: Value Menu Regexp: \`\*Echo Area
INS DEL Choice: Value Menu Regexp: \`\*Minibuf
INS DEL Choice: Value Menu Regexp: magit-.*: .*
INS
    State : CHANGED outside Customize.
   The regexp list that match boring buffers. More
Groups: Helm Buffers
```

<details><summary>
Windows 10, System Info (Click to expand)
</summary>
<pre>
#### System Info :computer:
- OS: windows-nt
- Emacs: 27.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. e4452f9a8)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(auto-completion autohotkey emacs-lisp git helm html multiple-cursors
                 (org :variables org-agenda-files
                      '("c:/Users/username/org/"))
                 shell spell-checking syntax-checking treemacs version-control)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY W32NOTIFY ACL GNUTLS LIBXML2 HARFBUZZ ZLIB TOOLKIT_SCROLL_BARS MODULES THREADS JSON PDUMPER LCMS2 GMP

</pre>
</details>

**Comment by hansbogert** [comment]
@duianto I'm not sure; are you confirming my problem, or giving a solution :relaxed:

**Comment by duianto** [comment]
It looked like you only had tried to change the helm boring buffer variable.
Because the screenshots following [`voidfoo`'s suggestion](https://github.com/syl20bnr/spacemacs/issues/12837#issuecomment-781970838), 
only showed the helm boring buffer variable in the customize buffer.

`voidfoo`'s suggestion worked for me. And when I tried to only change 
the helm boring buffer variable, then I saw what you described.

If you haven't tried `voidfoo` suggestion, then my comment 
explains why it doesn't work to set helm boring buffer variable directly.

Otherwise it just demonstrates what happened for me.
That means that we'll have to go bug hunting.
What's your system info, press `SPC h d s` to copy it to your clipboard.
